core/trino-main/src/test/java/io/trino/sql/TestExpressionInterpreter.java

```diff
@@ -518,6 +523,12 @@ public void testIn()
         assertOptimizedEquals("0 / 0 in (2, 4, 2, 4)", "0 / 0 in (2, 4)");
         assertOptimizedEquals("0 / 0 in (rand(), 2, 4)", "0 / 0 in (2, 4, rand())");
         assertOptimizedEquals("0 / 0 in (2, 2)", "0 / 0 = 2");
+
+        // should not create a new instance of InPredicate if IN list consisted of literals only
```

Comment:
You document what the test is asserting. Document also _why_ it's important.

Reply:
Isn't it sufficient to have description in main and commit message? Why to document it 3 times?